utils/cli/definition.py

```diff
@@ -0,0 +1,204 @@
+import argparse
+import getpass
+import json
+import logging
+import os
+from enum import Enum
+
+
+def get_auth(s):
+    """ Parse USER[:PASS] strings and prompt for password if none was
+    supplied. """
+    user, _, password = s.partition(':')
+    password = password or os.environ.get('PASS') or getpass.getpass()
+    return user, password
+
+
+def parse_label_arg(s):
+    """ If s is a file load it as JSON, otherwise parse s as JSON."""
+    if os.path.exists(s):
+        fp = open(s, 'r')
+        return json.load(fp)
+    else:
+        return json.loads(s)
+
+
+class ResourceType(Enum):
+
+    LOCAL = 0
+    SHARE = 1
+    REMOTE = 2
+
+    def __str__(self):
+        return self.name.lower()
+
+    def __repr__(self):
+        return str(self)
+
+    @staticmethod
+    def argparse(s):
+        try:
+            return ResourceType[s.upper()]
+        except KeyError:
+            return s
+
+
+#######################################################################
+# Command line interface definition
+#######################################################################
+
+parser = argparse.ArgumentParser(
+    description='Perform common operations related to CVAT tasks.\n\n'
+)
+task_subparser = parser.add_subparsers(dest='action')
+
+#######################################################################
+# Positional arguments
+#######################################################################
+
+parser.add_argument(
+    '--auth',
+    type=get_auth,
+    metavar='USER:[PASS]',
+    default=getpass.getuser(),
+    help='''defaults to the current user and supports the PASS
+            environment variable or password prompt
+            (default user: %(default)s).'''
+)
+parser.add_argument(
+    '--server-host',
+    type=str,
+    default='localhost',
+    help='host (default: %(default)s)'
+)
+parser.add_argument(
+    '--server-port',
+    type=int,
+    default='8080',
+    help='port (default: %(default)s)'
+)
+parser.add_argument(
+    '--debug',
+    action='store_const',
+    dest='loglevel',
+    const=logging.DEBUG,
+    default=logging.INFO,
+    help='show debug output'
+)
+
+#######################################################################
+# Create
+#######################################################################
+
+task_create_parser = task_subparser.add_parser(
+    'create',
+    description='Create a new CVAT task.'
+)
+task_create_parser.add_argument(
+    '--name',
+    default='new task',
+    type=str,
+    help='name of the task (default: %(default)s)'
+)
+task_create_parser.add_argument(
+    '--labels',
+    default='[]',
+    type=parse_label_arg,
+    help='string or file containing JSON labels specification'
+)
+task_create_parser.add_argument(
```

Comment:
What is about other attributes? I'm OK to have a minimal patch. Just curious if you like to add more parameters (images quality, owner, assignee, etc...). All of the can be viewed in http://localhost:8080/api/swagger documentation